src/pages/[...slug].tsx

```diff
@@ -16,7 +16,7 @@ export const getServerSideProps: GetServerSideProps<any> = async (context: GetSe
       theme = await getDefaultTheme(context.query.preview !== undefined);
     }
```

Comment:
nit: we can do the same for the `theme` as we do in the static routes:

```javascript
const theme = page.theme ?? (await getDefaultTheme(preview)) ?? {};
```

and then remove the `theme || null` from line 25